apps/ensapi/src/lib/ensanalytics/types.ts

```diff
@@ -0,0 +1,24 @@
+import type { AggregatedReferrerMetrics, Duration, UnixTimestamp } from "@ensnode/ensnode-sdk";
+
+/**
+ * Represents a snapshot of top referrers.
+ */
+export interface TopReferrersSnapshot {
+  /**
+   * Array of top referrers sorted by total incremental duration (descending)
+   * @invariant Array may be empty if there are no qualified referrers.
+   */
+  topReferrers: AggregatedReferrerMetrics[];
+  /** Unix timestamp of when the top referrers were last updated */
+  updatedAt: UnixTimestamp;
+  /**
+   * The sum of totalReferrals across all referrers
```

Comment:
```suggestion
   * @invariant The sum of totalReferrals across all referrers
```

I see this as a nice invariant too :)